modules/imgproc/test/test_distancetransform.cpp

```diff
@@ -283,4 +283,27 @@ void CV_DisTransTest::prepare_to_validation( int /*test_case_idx*/ )
 
 TEST(Imgproc_DistanceTransform, accuracy) { CV_DisTransTest test; test.safe_run(); }
 
+BIGDATA_TEST(Imgproc_DistanceTransform, large_image_12218)
+{
+    const int lls_maxcnt = 79992000;   // labels's maximum count
+    const int lls_mincnt = 1;           // labels's minimum count
+    int i, j, nz;
+    Mat src(8000, 20000, CV_8UC1), dst, labels;
+    for( i = 0; i < src.rows; i++ )
+        for( j = 0; j < src.cols; j++ )
+            src.data[i*src.step + j] = ( j > (src.cols / 2)) ? 0 : 255;
+
+    distanceTransform(src, dst, labels, cv::DIST_L2, cv::DIST_MASK_3, DIST_LABEL_PIXEL);
+
+    Size size = labels.size();
+    for( i = 0; i < size.height; i++ )
+    {
+        int* lls = labels.ptr<int>(i);
+        for( j = 0; j < size.width; j++ )
+            lls[j] = (lls[j] - lls_mincnt) / (lls_maxcnt / 2);
```

Comment:
Element-wise `B = A * alpha + beta` operation can be performed via `cv::convertTo()` call.